hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationFuture.java

```diff
@@ -164,7 +166,20 @@ private V resolveResponse() throws ExecutionException, TimeoutException, Interru
         if (response == null) {
             throw new TimeoutException();
         }
-        return (V) response;
+        return (V) verifyResponse(response);
+    }
+
+    private Object verifyResponse(Object value) {
+        if (value instanceof Data) {
+            return value;
+        }
+        if (operationResultVerifier != null) {
+            value = operationResultVerifier.verify((V) value);
```

Comment:
Because this OperationResultVerifier is a general addition and nothing JCache specific. Therefore it found it more useful to not throw exceptions directly but yeah maybe we should do it.